test/api.test.ts

```diff
@@ -0,0 +1,233 @@
+// import request from "supertest";
+import { expect, assert } from "chai";
+import sinon from "sinon";
+import request from "supertest";
+import App from "../src/app";
+import { adapters, StorageService } from "../src/storage";
+import {
+  BidPayloadPostDto,
+  BidsAccountsDto,
+  BidsListDto,
+  BidPostDto,
+  VerifyBidResponse,
+  Auction,
+  BidsDto,
+} from "../src/types";
+import * as util from "../src/util";
+import * as auctions from "../src/util/auctions";
+import * as contracts from "../src/util/contracts";
+
+describe("Test API Endpoints", async () => {
+  sinon.stub(contracts, "encodeBid").returns(Promise.resolve(""));
+
+  const stubbedStorageService = {
+    uploadFile: () => {},
+    downloadFile: () => {},
+    safeDownloadFile: (auctionFileKey: string) => {
+      return auctionFileKey;
+    },
+  };
+  sinon
+    .stub(adapters.fleek, "create")
+    .returns(
+      (await Promise.resolve(
+        stubbedStorageService
+      )) as unknown as StorageService
+    );
+  sinon
+    .stub(util, "getFleekConnection")
+    .returns(
+      (await Promise.resolve(
+        stubbedStorageService
+      )) as unknown as StorageService
+    );
+
+  const stubbedVerifyBidResponse = {
+    pass: true,
+    status: 200,
+    message: "OK",
+  };
+  sinon
+    .stub(auctions, "verifyEncodedBid")
+    .returns(
+      Promise.resolve(stubbedVerifyBidResponse)
+    ) as unknown as VerifyBidResponse;
+
+  const stubbedGetOrCreateAuction = {
+    tokenId: "0x123",
+    contractAddress: "0x456",
+    bids: [],
+  };
+  sinon
+    .stub(auctions, "getOrCreateAuction")
+    .returns(Promise.resolve(stubbedGetOrCreateAuction)) as unknown as Auction;
```

Comment:
I think the `as unknown as Auction;` is un-needed here since it's not having any actual effect. (it's casting the return of `.returns` not what's inside of it)

Reply:
Why is this one unnecessary but having `as unknown as StorageService` a few lines above is required by TypeScript (errors if I don't have it)?